lib/utils.js

```diff
@@ -519,7 +519,7 @@ function resolveReadPreference(parent, options) {
     throw new Error('No readPreference was provided or inherited.');
   }
 
-  return readPreference;
+  return typeof readPreference === 'string' ? new ReadPreference(readPreference) : readPreference;
```

Comment:
is this something we've been missing this whole time?

Reply:
Yep! We have a pretty awful track record with read preferences in this codebase. Ultimately nothing done here even matters much because `core` uses `translateReadPreference` to convert it to the right value _in place_ within `options` before e.g. server selection :man_facepalming: